public/js/index.js

```diff
@@ -5,13 +5,14 @@ var defaultExtraKeys = {
       cm.setOption("fullScreen", !cm.getOption("fullScreen"));
     },
     "Esc": function(cm) {
-      if (cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
+       if (cm.getOption('keyMap').substr(0, 3) === 'vim') return CodeMirror.Pass;
+       else if(cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
     },
     "Cmd-S": function () {
-        return CodeMirror.PASS
+        return CodeMirror.Pass
```

Comment:
If you want to use default behavior, `PASS` is not working.
`CodeMirror.PASS` return `undefined`
`CodeMirror.Pass` return `Object { }`

[CodeMirror User Manual](https://codemirror.net/doc/manual.html#keymaps)

>  A key handler function may return CodeMirror.Pass to indicate that it has decided not to handle the key, and other handlers (or the default behavior) should be given a turn.

Or I got it wrong.